core/txbuilder/submitter.go

```diff
@@ -0,0 +1,28 @@
+package txbuilder
+
+import (
+	"context"
+
+	"chain/core/rpc"
+	"chain/errors"
+	"chain/protocol/bc"
+)
+
+// Submitter submits a transaction to the generator so that it may
+// be confirmed in a block.
+type Submitter interface {
```

Comment:
Arguably this should be defined closer to where it's used, like at the top of finalize.go?